### ISSUE TITLE
Internal links not focussing

### ISSUE THREAD

**Issue by astroDimitrios** [astroDimitrios/rose#12]
In `02-help` there is a link env=my_env3 which should focus the my_env3 entry box but it isn't.

![image](https://github.com/user-attachments/assets/eac31e0e-35cd-4363-b60f-4cada0866654)
![image](https://github.com/user-attachments/assets/01a831a1-07ce-487d-abb3-4823713b9d85)

**Comment by J-J-Abram** [comment]
Linked to metomi/rose/config_editor/keywidget.py

Line 92:         self.grab_focus = self.entry.grab_focus

this allows the box entry to grab focus when you click on it


In metomi/rose/config_editor/variable.py

Line 388

```
    def grab_focus(self, focus_container=None, scroll_bottom=False,
                   index=None):
        """Method similar to Gtk.Widget - get the keyboard focus."""
        if hasattr(self, 'valuewidget'):
            self.valuewidget.grab_focus()
            if (index is not None and
                    hasattr(self.valuewidget, 'set_focus_index')):
                self.valuewidget.set_focus_index(index)
            for child in self.valuewidget.get_children():
                if (self.valuewidget.get_sensitive() & child.get_state_flags() and
                        self.valuewidget.get_parent().get_sensitive() & child.get_state_flags()):
                    break
            else:
                if hasattr(self, 'menuwidget'):
                    self.menuwidget.get_children()[0].grab_focus()
            if scroll_bottom and focus_container is not None:
                self.force_scroll(None, container=focus_container)
        if hasattr(self, 'keywidget') and self.key == '':
            self.keywidget.grab_focus()
        return False
```

**Comment by astroDimitrios** [comment]
For this page the problem is in `metomi/rose/config_editor/variable.py` `grab_focus()` and  `metomi/rose/config_editor/valuewidget/__init__.py` `get_focus()`.

I have removed the hook with a focus function and added in a check in `get_focus()` for Gtk.Entry and set the focus there. This highlights all the text on clicking the link (if this is not desired we can change to `Gtk.Entry.grab_focus_without_selecting(widget)`. For some reason the focus on the cog button which I have set to a gray outline doesn't show until you hit the tab key the first time in a session. Given we have spent a day on this I suggest this is fine for now and we move on.
